Makefile

```diff
@@ -19,26 +19,27 @@ INSTALLDIR=`$(OCAMLC) -where`/zip
 
 ### End of configuration section
 
-OCAMLC=ocamlc -g -safe-string
-OCAMLOPT=ocamlopt -safe-string
-OCAMLDEP=ocamldep
-OCAMLMKLIB=ocamlmklib
+OCAMLC=ocamlfind ocamlc -g -safe-string
+OCAMLOPT=ocamlfind ocamlopt -safe-string
+OCAMLDEP=ocamlfind ocamldep
+OCAMLMKLIB=ocamlfind ocamlmklib
 
 OBJS=zlib.cmo zip.cmo gzip.cmo
 C_OBJS=zlibstubs.o
 
-NATDYNLINK := $(shell if [ -f `ocamlc -where`/dynlink.cmxa ]; then echo YES; else echo NO; fi)
+include $(shell ocamlfind ocamlc -where)/Makefile.config
+NATDYNLINK := $(shell if [ -f `ocamlfind ocamlc -where`/dynlink.cmxa ]; then echo YES; else echo NO; fi)
```

Comment:
If we are to include OCaml's Makefile.config, there is probably a simpler way to determine whether .cmxs files are supported.